content/tutorials/spatial_standards_vector/index.Rmd

```diff
@@ -0,0 +1,225 @@
+  read_sf("sac.shp") %>% 
+  select(sac_code = GEBCODE,
+         sac_name = NAAM,
+         subsac_code = DEELGEBIED,
+         polygon_id = POLY_ID)
+```
+
+Have a look at its contents by printing the object:
+
+```{r paged.print=FALSE}
+sac
+```
+
+To write the GeoPackage, we just use the GPKG driver of the powerful [GDAL](https://gdal.org) library (supporting most open and some closed formats), which can be elegantly accessed through `sf::st_write()`:
+
+```{r}
+sac %>% 
+  st_write("sac.gpkg")
+```
+
+Is that all?\
+_**YES :-)**_
+
+Really?\
+_**YES :-)**_
+
+Well, hmmm, if you really want to know a little bit more...
+
+A GeoPackage can contain many layers.
+So, it is good practice to explicitly define the layer name within the GeoPackage (above, it was automatically called 'sac').
+For example:
+
+```{r}
+sac %>% 
+  st_write("sac.gpkg",
+           layer = "special_areas_conservation",
+           delete_dsn = TRUE)
+```
+
+Note, `delete_dsn` was set as `TRUE` to replace the whole GeoPackage.
+(There is also a `delete_layer` parameter to overwrite an existing _layer_ with the same name.)
+
+Let's extract a selection of features from the `special_areas_conservation` layer, and add it as a second layer into the GeoPackage file:
+
+```{r}
+sac %>% 
+  filter(str_detect(sac_name, "Turnhout")) %>% # only polygons in the Turnhout area
+  st_write("sac.gpkg",
+           layer = "turnhout")
+```
+
+So yes, adding layers to a GeoPackage is done simply by `st_write()` again to the same GeoPackage file (by default, `delete_dsn` is `FALSE`), and defining the new layer's name.
+
+So, which layers are available in the GeoPackage?
+
+```{r}
+gdalUtils::ogrinfo("sac.gpkg") %>% 
+  cat(sep = "\n")
+```
+
+_**You see?**_
+
+### Reading a GeoPackage file
+
+Can it become more simple than this?
+
+```{r}
+# (note: the 'layer' argument is unneeded if there's just one layer)
+sac_test <- st_read("sac.gpkg",
+                    layer = "special_areas_conservation")
+```
+
+Ready!
+
+Also other geospatial software will (or should be) able to open the GeoPackage format.
+It is an open standard, after all!
+
+
+## How to make and use GeoJSON files (`*.geojson`)
+
+### Making a GeoJSON file from a geospatial `sf` object in R
+
+As another example, let's download a shapefile of stream habitat 3260 in Flanders (version _2018_):
+
+```{r}
+download_zenodo(doi = "10.5281/zenodo.3386246")
+```
+
+_Again: you can visit a website of this dataset by just prefixing the DOI with `doi.org/`!_
+
+The data source is a shapefile again, in this case consisting of 4 different files.
+Similar as above, we read the geospatial data into R as an `sf` object and select a few attributes to work with:
+
+```{r}
+habitatstreams <- 
+  read_sf("habitatstreams.shp") %>% 
+  select(river_name = NAAM,
+         source = BRON)
+```
+
+
+```{r paged.print=FALSE}
+habitatstreams
+```
+
+Nowadays, it is recommended to use the more recent and strict **RFC7946** implementation of GeoJSON.
```

Comment:
How can you tell from the geoJSON file, which version is used?

Reply:
It is not hardcoded in the file which version is used, so it is not always clear. From the RFC7946 specification I understand it is a more constrained version of GeoJSON 2008, the most obvious being that WGS84 is obliged. RFC7946 does not contradict 2008, so some 2008 implementations may already conform to RFC7946 as well.

You can easily tell a GeoJSON file is _not_ RFC746 _when_ it has a `crs` key. The `crs` key is [absent](https://tools.ietf.org/html/rfc7946#appendix-B.1) from RFC7946. Other RFC7946 constraints will be less obvious to check from the file, maybe there are tools to check that.

There are also extra recommendations when applying RFC7946, but these are not obligations, so that's more difficult to rely on. A rather important one seems the [advice on coordinate precision](https://tools.ietf.org/html/rfc7946#section-11.2), which has effect in GDAL.

Hence, specifically for files generated with GDAL's default for these verions, it will be easier to distinguish both. GeoJSON 2008 by default [uses](https://gdal.org/drivers/vector/geojson.html#layer-creation-options) `COORDINATE_PRECISION=15` while it is `COORDINATE_PRECISION=7` for RFC7946.

IMHO a specific pitfall in GDAL's implementation of GeoJSON 2008 is that it will not transform data from another CRS to WGS84, while at the same time _not_ declaring `crs` in the file! (Unless when input already was WGS84, though.) Hence, when reading, WGS84 is assumed with wrong coordinates. Cf. [this post](https://github.com/r-spatial/sf/issues/344#issue-229118527).